python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+                average values
+            - donutQualityTable : astropy.table.QTable
+                Quality information for donuts
+
+        Notes
+        -----
+        This implementation processes a single exposure containing donut
+        stamps from either the intra-focal or extra-focal position.
+        The TARTS neural network is used to estimate Zernike coefficients
+        representing wavefront aberrations.
+
+        The exposure should contain:
+        - Valid image data (typically donut stamps)
+        - Proper WCS (World Coordinate System) information
+        - Appropriate metadata for the instrument and observation
+
+        See Also
+        --------
+        calcZernikesFromExposure : Method that processes individual exposures
+        """
+        # Check if exposure is valid
+        self.log.info("Starting Zernike estimation for a single exposure")
+        self.log.debug("RUN METHOD DEBUG: CalcZernikesNeuralTask.run() called")
+        if exposure is None:
+            # No exposure available - return empty results
+            self.log.warning("No exposure provided; returning empty results")
+            return self.empty()
+
+        # Store exposure for metadata creation
+        self._current_exposure = exposure
+
+        # Determine defocal type using helper function
+        try:
+            defocalType = self._determine_defocal_type(exposure)
+        except ValueError as e:
+            self.log.error("Failed to determine defocal type: %s", e)
+            # Raise exception to prevent incorrect processing
+            # In the future, this could be made configurable or handled
+            # differently
+            raise
+        pred, donutStamps, zk = self.calcZernikesFromExposure(exposure, defocalType)
+
+        # Log final donut count and processing summary
+        num_donuts_final = len(donutStamps) if donutStamps else 0
+        self.log.info("Final processing summary: %d donut(s) processed, defocalType='%s'",
+                     num_donuts_final, defocalType)
+        self.log.debug(
+            "Pred shape pre-squeeze: %s, donut stamps: %d, total zernikes shape: %s",
+            np.shape(pred),
+            len(donutStamps),
+            np.shape(zk),
+        )
+        # Convert PyTorch tensors to NumPy arrays
+        # pred and zk should already be numpy arrays
+        pred_np = pred if isinstance(pred, np.ndarray) else np.array(pred)
+        zk_np = zk if isinstance(zk, np.ndarray) else np.array(zk)
+
+        # Ensure pred is 1D array of Zernike coefficients
+        if pred_np.ndim > 1:
+            pred_np = pred_np[0, :]  # Take first row if multi-dimensional
+
+        # For outputZernikesRaw and outputZernikesAvg, use the same structure
+        # as standard tasks
+        # Both should be 2D arrays with shape (1, n_zernikes) for single
+        # exposure
+        zernikesRaw = np.atleast_2d(pred_np)  # Single prediction per exposure
```

Comment:
this should really be the predictions you get before your aggregatorNet, in my opinion, so the list of raw values per donut